tests/network/pod_networking.go

```diff
@@ -0,0 +1,115 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGDescribe("Pod Networking", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+	})
+
+	vmiWithDefaultBinding := func() *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.Spec.Domain.Devices.Interfaces = nil
+		vmi.Spec.Networks = nil
+		return vmi
+	}
+
+	vmiWithBridgeBinding := func() *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultBridgeNetworkInterface()}
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+		return vmi
+	}
+
+	vmiWithMasqueradeBinding := func() *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+		return vmi
+	}
+
+	vmiWithBinding := func(bindingMechanism string) *v1.VirtualMachineInstance {
+		switch bindingMechanism {
+		case "bridge":
+			return vmiWithBridgeBinding()
+		case "masquerade":
+			return vmiWithMasqueradeBinding()
+		case "default":
+			return vmiWithDefaultBinding()
+		default:
+			panic(fmt.Sprintf("Selected binding %q is invalid", bindingMechanism))
+		}
+	}
+
+	testStatusReporting := func(description string, bindingMechanism string) {
+		Context(fmt.Sprintf("VMI connected to the pod network using %s", description), func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				By("Creating the VMI")
+				var err error
+				vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmiWithBinding(bindingMechanism))
+				Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+
+				By("Waiting until the VMI gets ready")
+				vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInAlpineExpecter)
+			})
+
+			AfterEach(func() {
+				By("Deleting the VMI")
+				err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})
```

Comment:
When only the error is returned, consider using the Success matcher: `Expect(Foo()).To(Succeed())`, then you do not need to explain with text and it reads more natural.

Reply:
Done